diff --git a/src/aspforaba/encoder.py b/src/aspforaba/encoder.py
index a866c76..4e065f2 100644
--- a/src/aspforaba/encoder.py
+++ b/src/aspforaba/encoder.py
@@ -40,7 +40,10 @@ class Encoder:
 
     COMPLETE_ADD = ":- out(X), not attacked_by_undefeated(X)."
 
-    STABLE_ADD = ":- not defeated(X), out(X)."
+    STABLE_ADD = """
+        :- assumption(X), supported(X), not in(X).
+        :- not defeated(X), out(X).
+        """
 
     GROUNDED_ENCODING = """
         n_assumptions(N) :- #count{X : assumption(X)} = N.
