diff --git a/causalaba.py b/causalaba.py
index 50a34db..bfe2fb8 100644
--- a/causalaba.py
+++ b/causalaba.py
@@ -26,6 +26,8 @@ from tqdm.auto import tqdm
 from itertools import combinations
 from datetime import datetime
 from utils.graph_utils import powerset, extract_test_elements_from_symbol
+from pathlib import Path
+
 
 def compile_and_ground(n_nodes:int, facts_location:str="",
                 skeleton_rules_reduction:bool=False,
@@ -51,7 +53,7 @@ def compile_and_ground(n_nodes:int, facts_location:str="",
             # logging.debug(f"in({s},{'s'+'y'.join([str(i) for i in S])}).")
 
     ### Load main program and facts
-    ctl.load("encodings/causalaba.lp")
+    ctl.load(str(Path(__file__).resolve().parent / 'encodings' / 'causalaba.lp'))
     if facts_location != "":
         ctl.load(facts_location)
     if weak_constraints:
